elastic/eui
# [EuiComboBox] Replace autosizing input dependency with more performant utility

## Summary

closes https://github.com/elastic/eui/issues/4866, and incidentally partially (but not completely) addresses https://github.com/elastic/eui/issues/7080

When I was upgrading `react-autosizer-input` with React 18 support, I started grepping through to see usages and I was like... "Hmm, it's just one usage _just_ for EuiComboBox? This isn't a hard problem to solve, surely we could just roll our own instead..." and it turns out not only can we Do It Ourselves, we can utilize our new canvas rendering text utils to do it way more performantly 🎉 

> ℹ️ Quick note, this PR has extra reported diffs simply due to spacing changes - I'd recommend [viewing diffs with whitespace hidden](https://github.com/elastic/eui/pull/7215/files?diff=split&w=1).

## QA

- Go to https://eui.elastic.co/pr_7215/#/forms/combo-box
- Confirm that typing in search text (including very long search text) works the [same as production](https://eui.elastic.co/v88.4.1/#/forms/combo-box)

### General checklist

- Browser QA
- Docs site QA - N/A
- Code quality checklist
- Release checklist
